pkg/steps/cluster_claim.go

```diff
@@ -61,33 +61,59 @@ func (s *clusterClaimStep) run(ctx context.Context) error {
 		// should never happen
 		return fmt.Errorf("cannot claim a nil cluster")
 	}
-	clusterClaim, err := acquireCluster(ctx, *s.clusterClaim, s.hiveClient, s.client, *s.jobSpec)
+	waitForClaim := func(client ctrlruntimeclient.WithWatch, ns, name string, claim *hivev1.ClusterClaim, timeout time.Duration) error {
+		logrus.WithFields(logrus.Fields{
+			"namespace": ns,
+			"name":      name,
+		}).Trace("Waiting for claim to be running.")
+
+		ctx := context.TODO()
+
+		evaluatorFunc := func(obj runtime.Object) (bool, error) {
+			switch clusterClaim := obj.(type) {
+			case *hivev1.ClusterClaim:
+				// https://github.com/openshift/hive/blob/a535d29c4d2dc4f7f9bf3f30098c69d9334bee2e/apis/hive/v1/clusterclaim_types.go#L18-L22
+				for _, condition := range clusterClaim.Status.Conditions {
+					if condition.Type == hivev1.ClusterRunningCondition && condition.Status == corev1.ConditionTrue {
+						return true, nil
+					}
+				}
+			default:
+				return false, fmt.Errorf("clusterClaim/%v ns/%v got an event that did not contain a clusterClaim: %v", name, ns, obj)
+			}
+			return false, nil
+		}
+
+		return waitForConditionOnObject(ctx, client, ctrlruntimeclient.ObjectKey{Namespace: ns, Name: name}, &hivev1.ClusterClaimList{}, claim, evaluatorFunc, timeout)
+	}
+
+	clusterClaim, err := s.acquireCluster(ctx, waitForClaim)
 	if err != nil {
 		acquireErr := results.ForReason("acquiring_cluster_claim").ForError(err)
 		// always attempt to delete claim if one exists
 		var releaseErr error
 		if clusterClaim != nil {
-			releaseErr = results.ForReason("releasing_cluster_claim").ForError(releaseCluster(ctx, s.hiveClient, clusterClaim, s.Name()))
+			releaseErr = results.ForReason("releasing_cluster_claim").ForError(s.releaseCluster(ctx, clusterClaim))
 		}
 		return aggregateWrappedErrorAndReleaseError(acquireErr, releaseErr)
 	}
 
 	wrappedErr := results.ForReason("executing_test").ForError(s.wrapped.Run(ctx))
-	releaseErr := results.ForReason("releasing_cluster_claim").ForError(releaseCluster(ctx, s.hiveClient, clusterClaim, s.Name()))
+	releaseErr := results.ForReason("releasing_cluster_claim").ForError(s.releaseCluster(ctx, clusterClaim))
 
 	return aggregateWrappedErrorAndReleaseError(wrappedErr, releaseErr)
 }
 
-func acquireCluster(ctx context.Context, clusterClaim api.ClusterClaim, hiveClient ctrlruntimeclient.Client, client loggingclient.LoggingClient, jobSpec api.JobSpec) (*hivev1.ClusterClaim, error) {
+func (s *clusterClaimStep) acquireCluster(ctx context.Context, waitForClaim func(client ctrlruntimeclient.WithWatch, ns, name string, claim *hivev1.ClusterClaim, timeout time.Duration) error) (*hivev1.ClusterClaim, error) {
```

Comment:
If `acquireCluster` is now a method on `*clusterClaimStep` why do we construct the waiting func outside of this and pass it in?